app/src/main/java/mozilla/lockbox/presenter/AuthPresenter.kt

```diff
@@ -52,33 +55,38 @@ class AuthPresenter(
 
         lockedStore.onAuthentication
             .subscribe {
-                if (it is FingerprintAuthAction.OnAuthentication) {
-                    when (it.authCallback) {
-                        is FingerprintAuthCallback.OnAuth -> unlock()
-                        is FingerprintAuthCallback.OnError -> view.unlockFallback()
+                when (it) {
+                    is FingerprintAuthAction.OnAuthentication -> {
+                        when (it.authCallback) {
+                            is FingerprintAuthCallback.OnAuth -> unlock()
+                            is FingerprintAuthCallback.OnError -> unlockFallback()
+                        }
                     }
+                    is FingerprintAuthAction.OnCancel -> finishResponse()
                 }
             }
             .addTo(compositeDisposable)
 
         view.unlockConfirmed
-            .filter { it }
             .subscribe {
-                unlock()
+                if (it) {
+                    unlock()
+                } else {
+                    finishResponse()
```

Comment:
Should this be followed by a Lock action?

Reply:
at this stage, we know that `DataStore` is still definitely locked.  I'm hesitant to add a (re) lock given how async these flows end up being.